frontend/src/utils/store-sync.ts

```diff
@@ -47,7 +47,7 @@ export async function fetchSettingsFromServer<T>(
   key: string,
   target: T,
   cast = true
-): Promise<T | DisplayPreferencesDto> {
+): Promise<DisplayPreferencesDto> {
```

Comment:
After removing all the typecasts, this function always return `DisplayPreferencesDto`, so it just worked out nicely

Reply:
If data is casted, definitely not ``DisplayPreferencesDto`` is returned here

Reply:
@aweebs No, the typings are correct. However, when we do the cast, we're turning them to native types.

[Server response (``DisplayPreferencesDto``)](https://api.jellyfin.org/#tag/DisplayPreferences):

```
{
  ...whatever,
  CustomPrefs: {
    hello: "0",
    test: "null"
  }
}
```

When cast is true we turn that into:
```
{
  ...whatever,
  CustomPrefs: {
    hello: 0,
    test: null
  }
}
```

Effectively matching the store's native types.
Do note the difference between TypeScript casting and runtime castings. Here we're doing a runtime casting.

Reply:
The current return value of `castDisplayPreferencesResponse` is typed as `DisplayPreferencesDto`, but the return value in the function is `data.CustomPrefs || {}` on line 37. Currently this function will pick certain properties off `DisplayPreferencesDto#CustomPrefs` and return that object, which is explicitly not `DisplayPreferencesDto`. In practice it does return type `T`. The reason there is no type error right now is because every property of `DisplayPreferencesDto` is optional which allows any object type to successfully check against it.